### PR TITLE
Adding QR Code Reader

### PR DESCRIPTION
**Pull request by macintosh1398**
Adding a new feature, which is the QR Code reader. A new main screen button give the user the ability to open a QR-Code scanner. This QR-Code scanner contains the basic camera feature and is able to display the content on the screen. The old PR has been closed due to the huge amount of conflicts with the main branch.

### CODE REVIEW

**Comment by macintosh1398** [comment]
The old pull request request has been closed due to the large number of conflicts that I was not able to solve. It was easier for me to do a fresh clone and add my changes.

**Review by Voxanimus — COMMENTED** [review]
Which framework did you use to testing camera ? only Espresso or also roboelctric ?
I did not very understood on which element is set the onClickListener.

**Comment by macintosh1398** [comment]
For the tests I only used Espresso for now (since last week Robolectric was not taken into account for the coverage). The QR-Code reading in itself was also tested manually. But I've planned to add Robolectric tests in the future.

File: app/src/main/java/com/github/multimatum_team/multimatum/MainActivity.kt [hunk 1]
```diff
@@ -20,6 +23,23 @@ class MainActivity : AppCompatActivity() {
         startActivity(intent)
     }
 
+    fun openCodeScanner(view: View) {
```
Comment (ValentinAebi) [inline app/src/main/java/com/github/multimatum_team/multimatum/MainActivity.kt:26]: If I understand correctly, if permission is denied the function asks for the permission and then returns. Wouldn't it be better to also start the QR code reader once permission has been granted (i.e. move lines 38-39 out of the `else` statement)?

File: app/src/androidTest/java/com/github/multimatum_team/multimatum/QRCodeReaderActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            ViewAssertions.matches(
+                isNotSelected()
+            )
+        )
+        Intents.release()
+    }
+
+    @get:Rule
+    val permissionRule: GrantPermissionRule = GrantPermissionRule.grant(
+        android.Manifest.permission.CAMERA
```
[the remark this answers is not in the document]
Comment (macintosh1398) [reply app/src/androidTest/java/com/github/multimatum_team/multimatum/QRCodeReaderActivityTest.kt:30]: Done

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -61,5 +61,19 @@
         app:layout_constraintStart_toStartOf="parent"
         tools:ignore="HardcodedText" />
 
+    <Button
+        android:id="@+id/goToQrCodeReader"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:onClick="openCodeScanner"
+        android:text="Scanner"
```
[the remark this answers is not in the document]
Comment (macintosh1398) [reply app/src/main/res/layout/activity_main.xml:69]: Done

**Comment by macintosh1398** [comment]
Note: as done for the QR generator, a return button is a good idea. For now, the Android return button is necessary but I will add it next week. The camera permission request does not launch the QR Code reading activity automatically for now (pressing again is necessary). It will be changed next week.

**Review by ValentinAebi — APPROVED** [review]

**Review by Voxanimus — APPROVED** [review]

**Pull request merged**